core/src/dird/testfind.cc

```diff
@@ -26,29 +26,78 @@
  */
 
 #include "include/bareos.h"
-#include "dird/dird.h"
+#include "dird/dird_conf.h"
 #include "findlib/find.h"
 #include "lib/mntent_cache.h"
 #include "include/ch.h"
 #include "filed/fd_plugins.h"
 #include "lib/parse_conf.h"
-#include "dird/job.h"
+#include "dird/jcr_util.h"
 #include "dird/dird_globals.h"
 #include "dird/dird_conf.h"
 #include "dird/jcr_private.h"
 #include "lib/recent_job_results_list.h"
 #include "findlib/attribs.h"
 #include "filed/fileset.h"
 #include "lib/util.h"
+#include "lib/tree.h"
 
 #if defined(HAVE_WIN32)
 #  define isatty(fd) (fd == 0)
 #endif
 
 namespace directordaemon {
+
+
 bool DoReloadConfig() { return false; }
 }  // namespace directordaemon
 
+void TestfindFreeJcr(JobControlRecord* jcr)
```

Comment:
This is a doubling of DirdFreeJcr() which should be avoided.
Probably most of the functionality here is not required at all. 
Please strip down this function to a absolute minimum so that only things are freed that have been initialized before.